pkg/controller/backupbucket/actuator.go

```diff
@@ -66,7 +66,12 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, backupBucket *e
 			return util.DetermineError(err, helper.KnownCodes)
 		}
 
-		storageDomain, err := azureclient.BlobStorageDomainFromCloudConfiguration(backupConfig.CloudConfiguration)
+		bucketCloudConfiguration, err := azureclient.CloudConfiguration(backupConfig.CloudConfiguration, &backupBucket.Spec.Region)
```

Comment:
Can't we reuse `azCloudConfiguration` from line 45?

Reply:
No, that is an `cloud.Configuration` (azure SDK type) whereas this function requires a `azure.CloudConfiguration` (our type).

This is in part due to the way these types are defined in the sdk, which makes them rather unwieldy in general. Still, when interacting directly with the SDK clients (as in l.45, kind of) they are the way to go. But in other instances, our Type is much more reasonable to check against.